### DISCUSSION TITLE
正常采集过程中不定时发生日志重采

### DISCUSSION THREAD

**Discussion by apothecary-ap** [alibaba/ilogtail#1275] [Help] [answered]
在集群环境使用deamonset启动ilogtail,日志量相同，采集配置相同的集群中，发现单独一台ilogtail容器内存比其它内存高，日志采集重复
正常采集机器ilogtail只使用了100-300mb的内存，异常这台内存打满，一开始分配内存1g打满，修改分配2g还是打满
cpu无异常cpu使用都是0.3c-0.5c

ilogtial配置了多个采集配置，ilogtail_config.json为通用配置，以下只列了发生重采的采集配置
使用的ilogtail的版本是1.3.1
<details>
<summary>ilogtail_config.json</summary><pre>
{
    "cpu_usage_limit": 1,
    "mem_usage_limit": 2048,
    "polling_ignore_file_modify_timeout": 7200,
    "polling_file_first_watch_timeout": 600,
    "check_point_dump_interval": 5
}<pre></details>

<details>
<summary>ilogtail采集配置</summary><pre>
enable: true
inputs:
- Type: file_log
  LogPath: /logs/job-manager/serviceLog/
  FilePattern: '*.opsMonitorLog.*.log'
  EnableLogPositionMeta: true
processors:
- Type: processor_add_fields
  Fields:
    logtype: generallog
    domain: aiops
    servicename: aiops-demo
    agenthost: localhost
    ilogtail: ilogtail
    env: normal
flushers:
- Type: flusher_kafka_v2
  Compression: gzip
  Brokers:
  - XX.XX.XXX.XXX:XXXX
  Convert:
    ProtocolFieldsRename:
      contents: fields
  Topic: demo<pre></details>

<details>
<summary>重采发现的日志生成方式：</summary><pre>
每分钟打印20条数据，按小时进行文件切分
文件命名方式：xxx.日期-小时.log（jobmanager.opsMonitorLog.2023-12-19-06.log）<pre></details>

以下为部分debug日志：

发生重采的时候都有unregister
[ilogtail.LOG](https://github.com/alibaba/ilogtail/files/13735169/ilogtail.LOG)

**Comment by linrunqi08** [comment, marked as answer]
内存如果满了，会导致logtail进程被oom kill，内存中的checkpoint无法记录下来，从而导致重复

**Reply by apothecary-ap** [reply]
请问这种情况是怎么触发的，看debug日志没看出来

**Reply by apothecary-ap** [reply]
是我们ilogtail使用上有什么问题吗，很多台机器相同的配置就只有这一台发生这种现象

**Reply by apothecary-ap** [reply]
这是容器部署而且是单独分配的资源，为什么会出现oom的现象，如何修复避免这种情况再次出现

**Reply by linrunqi08** [reply]
这台机器的数据量如果很大的话，会引起内存升高的

**Reply by apothecary-ap** [reply]
有三台机器的日志量和这台一样达到了一天1亿-1.5亿条日志，但是三台机器的内存只用到了300mb不到，都是一样的配置

**Reply by linrunqi08** [reply]
这个不太合理，如果都是一样的话，不太可能就这台内存升高。可以用工具看一下mem profile，看看内存中什么比较高

**Reply by apothecary-ap** [reply]
多谢大佬的回复，这个内存排查mem profile有文档推荐参照吗，不了解怎么去排查，主要是做java开发的，只了解基础的c++和golang语法

**Reply by 7y-9** [reply]
ilogtail 插件中提供dump mem和cpu等接口，可以看下这个plugin_main/plugin_http.go
